### PR TITLE
Fix Compact User Dropdown

### PR DESCRIPTION
**Pull request by Tacodude7729**
Fixed two issues retron pointed out with the compact user dropdown setting in Customizable navigation bar while I was streaming on Discord.

The compactification not working on some pages, like the 'my suff' page.
image

Extra space after the profile icon on some pages that isn't present on the homepage.
image

### CODE REVIEW

**Comment by retronbv** [comment]
ty taco

**Review by apple502j — APPROVED** [review]

File: addons/discuss-button/r2.js [hunk 1]
```diff
@@ -5,8 +5,9 @@ export default async function ({ addon }) {
   const add = async () => {
     if (!span && (await addon.auth.fetchIsLoggedIn())) {
       const username = await addon.auth.fetchUsername();
-      const container = document.querySelector(".dropdown");
-      const dropdown = container.querySelector(".dropdown-menu .user-nav");
+      const container = await addon.tab.waitForElement(".dropdown");
+      const dropdown = await addon.tab.waitForElement(".dropdown-menu .user-nav");
+      (await addon.tab.waitForElement(".user-icon")).style = "margin-right: 0px;";
```
Comment (Samq64) [inline addons/discuss-button/r2.js:10]: Is there a reason for the extra set of parenthacies?

**Review by Samq64 — COMMENTED** [review]
Dynamically disabling the compact user dropdown removes the margin between the username and the icon.

File: addons/discuss-button/r2.js (hunk 1 above)
Comment (mxmou) [reply addons/discuss-button/r2.js:10]: I think the browser would read it as
```js
(await addon.tab.waitForElement(".user-icon").style) = "margin-right: 0px;";
```
without them.

**Review by WorldLanguages — DISMISSED** [review]
Bug reported by Samq64 above

**Comment by Tacodude7729** [comment]
@WorldLanguages Resolved the issue above

**Review by Samq64 — APPROVED** [review]

**Pull request merged**